libraries/stdlib/src/kotlin/ranges/PrimitiveRanges.kt

```diff
@@ -115,7 +115,7 @@ public class LongRange(start: Long, endInclusive: Long) : LongProgression(start,
         first == other.first && last == other.last)
 
     override fun hashCode(): Int =
-        if (isEmpty()) -1 else (31 * (first xor (first ushr 32)) + (last xor (last ushr 32))).toInt()
+        if (isEmpty()) -1 else 31 * (first.hashCode() + last.hashCode())
```

Comment:
It seems you have changed the formula here. It is worth adding a test for hash code before the change and ensure that it passes after.